pyproject.toml

```diff
@@ -12,30 +12,30 @@ authors = [
 description = "Kedro helps you build production-ready data and analytics pipelines"
 requires-python = ">=3.7"
 dependencies = [
-    "anyconfig~=0.10.0",
-    "attrs>=21.3",
-    "build",
-    "cachetools~=5.3",
-    "click<9.0",
-    "cookiecutter>=2.1.1, <3.0",
-    "dynaconf>=3.1.2, <4.0",
-    "fsspec>=2021.4, <2024.1",  # Upper bound set arbitrarily, to be reassessed in early 2024
-    "gitpython~=3.0",
-    "importlib-metadata>=3.6; python_version >= '3.8'",
-    "importlib_metadata>=3.6, <5.0; python_version < '3.8'",  # The "selectable" entry points were introduced in `importlib_metadata` 3.6 and Python 3.10. Bandit on Python 3.7 relies on a library with `importlib_metadata` < 5.0
-    "importlib_resources>=1.3",  # The `files()` API was introduced in `importlib_resources` 1.3 and Python 3.9.
-    "jmespath>=0.9.5, <2.0",
-    "more_itertools>=9,<11",
-    "omegaconf~=2.3",
-    "parse~=1.19.0",
+    "anyconfig>=0.10.0,<1.0.0",
+    "attrs>=21.3,<24.0",
+    "build>=0.7.0,<1.0.0",
+    "cachetools>=4.1,<6.0",
+    "click>=7.0,<9.0",
+    "cookiecutter>=2.1.1,<3.0",
+    "dynaconf>=3.1.2,<4.0",
+    "fsspec>=2021.4,<2024.1",  # Upper bound set arbitrarily, to be reassessed in early 2024
+    "gitpython>=3.0,<4.0",
+    "importlib_metadata>=3.6,<5.0; python_version < '3.8'",  # The "selectable" entry points were introduced in `importlib_metadata` 3.6 and Python 3.10. Bandit on Python 3.7 relies on a library with `importlib_metadata` < 5.0
+    "importlib-metadata>=3.6,<7.0; python_version >= '3.8'",
+    "importlib_resources>=1.3,<7.0",  # The `files()` API was introduced in `importlib_resources` 1.3 and Python 3.9.
+    "jmespath>=0.9.5,<2.0.0",
+    "more_itertools>=8.14.0,<11",
+    "omegaconf>=2.1.1,<3.0.0",
+    "parse>=1.19.0,<=2.0.0",
     "pip-tools>=6.5,<8",
-    "pluggy~=1.0",
-    "PyYAML>=4.2, <7.0",
-    "rich>=12.0, <14.0",
-    "rope>=0.21, <2.0",  # subject to LGPLv3 license
-    "setuptools>=65.5.1",
-    "toml~=0.10",
-    "toposort~=1.5",  # Needs to be at least 1.5 to be able to raise CircularDependencyError
+    "pluggy>=1.0,<2.0",
```

Comment:
This is essentially equivalent if I'm not mistaken, should we leave it as it is?